File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.py
def accept(s):
    a = ['a','e','i','o','u']
    a_count = 0
    A,B = 0,0
    for i in range(len(s)):
        if s[i] in a:
            a_count += 1
            A += 1
            B = 0
        else:
            A = 0
            B += 1
        if A == 3 or B == 3:
            return False
        if i<len(s)-1 and (s[i] == s[i+1]):
            if s[i] != "e" and s[i] != 'o':
                return False
    if a_count==0:
        return False
    return True
    


while True:
    s = input()
    if s == "end":
        break
    if accept(s):
        print(f"<{s}> is acceptable.")
    else:
        print(f"<{s}> is not acceptable.")

File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        while (true) {
            String line = br.readLine();
            if (line.equals("end")) break;
            boolean answer = true;
            char before = ' ';
            int a_count = 0; //모음 개수
            int b_count = 0; //자음 개수
            int cnt = 0; //전체 모음 개수
            String tmp = "aeiou";
            char[] t = tmp.toCharArray();
            for (char c : line.toCharArray()){
                boolean flag = false;
                for (char tt : t) {
                    if (c == tt) {
                        cnt += 1;
                        b_count = 0;
                        a_count += 1;
                        flag = true;
                    }
                }  if (!flag) {
                    b_count += 1;
                    a_count = 0;
                }
                if (a_count>=3 || b_count>=3) {
                    answer = false;
                    break;
                }
                if (before == c) {
                    if (c == 'e' || c=='o') continue;
                    else {
                        answer = false;
                        break;
                    }
                }
                before = c;

            }
            if (cnt<1) answer = false;

            if (answer) System.out.println("<"+line+"> is acceptable.");
            else System.out.println("<"+line+"> is not acceptable.");
        }
    }
}